src/test/kotlin/com/lykke/trade/volume/monitoring/service/process/AssetVolumeConverterTest.kt

```diff
@@ -47,33 +46,87 @@ class AssetVolumeConverterTest {
+        converter.convert("Asset3", BigDecimal.ONE, setOf("CrossAsset1", "CrossAsset2", "CrossAsset3"), "Asset2")
+    }
+
+    @Test
+    fun testConvertUnknownAssetPairThroughCrossAsset() {
+        Mockito.`when`(assetPairsHolder.getAssetPair("Asset3", "CrossAsset2"))
+                .thenReturn(AssetPair("CrossPair2-1", "Asset3", "CrossAsset2", 1))
+        Mockito.`when`(assetPairsHolder.getAssetPair("CrossAsset2", "Asset2"))
+                .thenReturn(AssetPair("CrossPair2-2-WithoutPrice", "Asset2", "CrossAsset2", 2))
+
+        Mockito.`when`(assetPairsHolder.getAssetPair("Asset3", "CrossAsset3"))
+                .thenReturn(AssetPair("CrossPair3-1", "Asset3", "CrossAsset3", 3))
+        Mockito.`when`(assetPairsHolder.getAssetPair("CrossAsset3", "Asset2"))
+                .thenReturn(AssetPair("CrossPair3-2", "Asset2", "CrossAsset3", 4))
+
+        Mockito.`when`(pricesHolder.getPrice("CrossPair2-1"))
+                .thenReturn(BigDecimal("1.1"))
+        Mockito.`when`(pricesHolder.getPrice("CrossPair3-1"))
+                .thenReturn(BigDecimal("10.123"))
+        Mockito.`when`(pricesHolder.getPrice("CrossPair3-2"))
+                .thenReturn(BigDecimal("15.6789"))
+
+        val result = converter.convert("Asset3", BigDecimal("1.23"), setOf("CrossAsset1", "CrossAsset2", "CrossAsset3"), "Asset2")
+        assertEquals(BigDecimal("0.79"), result)
     }
 
     @Test(expected = ApplicationException::class)
     fun testUnknownPriceToConvert() {
-        converter.convert("Asset1", BigDecimal.ONE, "Asset3")
+        converter.convert("Asset1", BigDecimal.ONE, emptySet(), "Asset3")
+    }
+
+    @Test
+    fun testConvertUnknownPriceThroughCrossAsset() {
+        Mockito.`when`(assetsHolder.getAsset("Asset3"))
+                .thenReturn(Asset("Asset3", 3))
+
+        Mockito.`when`(assetPairsHolder.getAssetPair("Asset1", "CrossAsset2"))
+                .thenReturn(AssetPair("CrossPair2-1", "Asset1", "CrossAsset2", 1))
+        Mockito.`when`(assetPairsHolder.getAssetPair("CrossAsset2", "Asset3"))
+                .thenReturn(AssetPair("CrossPair2-2-WithoutPrice", "Asset3", "CrossAsset2", 2))
+
+        Mockito.`when`(assetPairsHolder.getAssetPair("Asset1", "CrossAsset3"))
+                .thenReturn(AssetPair("CrossPair3-1", "Asset1", "CrossAsset3", 3))
+        Mockito.`when`(assetPairsHolder.getAssetPair("CrossAsset3", "Asset3"))
+                .thenReturn(AssetPair("CrossPair3-2", "Asset3", "CrossAsset3", 4))
+
+        Mockito.`when`(pricesHolder.getPrice("CrossPair2-1"))
+                .thenReturn(BigDecimal("1.1"))
+        Mockito.`when`(pricesHolder.getPrice("CrossPair3-1"))
+                .thenReturn(BigDecimal("10.123"))
+        Mockito.`when`(pricesHolder.getPrice("CrossPair3-2"))
+                .thenReturn(BigDecimal("15.6789"))
+
+        val result = converter.convert("Asset1", BigDecimal("1.23"), setOf("CrossAsset1", "CrossAsset2", "CrossAsset3"), "Asset3")
```

Comment:
suggest using `sortedSet` to preserve input order

Reply:
made config value `crossAssetIds` a list as order is important